### PR TITLE
add support for facebook

### LINKED ISSUE #51: 💡 [REQUEST] - Support Other Popular Social Media Sites
### Reference Issues

_No response_

### Summary

This is a global tracking issue for adding support for other social media sites. Currently, we are only aiming to support Reddit, but eventually adding support for:
- [x] Twitter  - #81 
- [x] Facebook ([no OCR required](https://user-images.githubusercontent.com/16009897/192125837-5f43db85-5eff-4a21-83fc-f8eb31e8279a.png)) - #107
- [x] Instagram (Browser) - #80 

Would be very ideal.

### Basic Example

N/a

### Drawbacks

- More work to maintain
- Differences in implementation will require careful structuring of code to ensure efficacy across multiple sites.

### Unresolved questions

_No response_

### LINKED ISSUE #115: [BUG] - Facebook support does not blur both description and image
### Description

When detecting the description has the blocked word but not the image alt text, it only blurs the description text. When detecting that the image alt text has the blocked word but not the description, it only blurs the image. We want both image and description text to be blurred because they are likely to be related (and therefore unwanted).

### Reproduction URL

_No response_

### Reproduction steps

```bash
1. Install the extension.
2. Turn on the Facebook toggle.
3. Go to https://www.facebook.com/pizzahutnz/.
4. Find a post that contains the word "pizza" in either the image alt text or description (but not both).
5. Only one of (the image or description) will be blurred.
```

### PR DESCRIPTION
**Pull request by Kanelmis**
Add support for Facebook following the same logic as `twitter.js`.  In addition, for Facebook, we support checking posts' text and the image alt. 

## Todo
- [x] support blurring post 
- [x] support blurring image

## Motivation
See issue https://github.com/se310-t6/politicry/issues/51. The aim is to support multiple social media platforms

## Attached GitHub issue
Closes https://github.com/se310-t6/politicry/issues/51. Closes #115

### Documentation

- [x] Updated the readme documents (`README.md` etc.)
- [x] New functions and methods have additional comments added to document their usage

### Local Build

- [x] Ran all pre-commit hooks `pre-commit run --all-files`
- [x] Extension has been tested to build correctly `cd extension && yarn && yarn build`
- [x] Backend has been tested to build correctly `cd backend && docker-compose build`
- [x] Extension test suite has been run locally `cd extension && yarn && yarn test`
- [x] Backend test suite has been run locally `cd backend && python3 -m unittest`

### GitHub

- [x] Target branch has been set correctly
- [x] PR has been rebased onto target branch
- [x] PR has been assigned an owner
- [x] Author has performed a self-review of the code
- [x] Removed the WIP message from the top of this PR

### CODE REVIEW

**Review by Destiny-02 — DISMISSED** [review]
Nice work! Tested and works :)
image

File: extension/public/worker-test.html [hunk 1]
```diff
@@ -64,6 +64,16 @@ <h1>Posts with allowed words</h1>
   <div lang="en">🟢 This post is about Pizza and Chicken 🟢</div>
   <div lang="en">🟢 This post is about Chicken 🟢</div>
 </main>
+
+<main id="facebook">
+  <h1>Facebook</h1>
+  <article>
+    <div lang="en">🔴 This tweet is about Pizza 🔴</div>
```
Comment (k-yle) [inline extension/public/worker-test.html]:
```suggestion
    <div lang="en">🔴 This post is about Pizza 🔴</div>
```

File: extension/public/worker-test.html [hunk 2]
```diff
@@ -64,6 +64,16 @@ <h1>Posts with allowed words</h1>
...
+  </article>
+  <article>
+    <div lang="en">🟢 This tweet is about Sushi 🟢</div>
```
Comment (k-yle) [inline extension/public/worker-test.html]:
```suggestion
    <div lang="en">🟢 This post is about Sushi 🟢</div>
```

File: extension/public/worker/facebook.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+document.addEventListener("politicry-ready", () => {
+// quit if facebook is not enabled
+    if (!window.enabled.facebook) return;
+  
+    // check every 1 second if a new post has been loaded
+    setInterval(() => {
+      // check every visible facebook post, which can be identified since it's a
+      // div with an attribute called lang="en". Politicry only supports English (for now)
```
Comment (k-yle) [inline extension/public/worker/facebook.js]:
```suggestion
      // div with an attribute called data-ad-preview="message"
```

**Review by k-yle — CHANGES_REQUESTED** [review]
nice work! i've left a few minor suggestions

**Review by k-yle — APPROVED** [review]
looks good

**Pull request merged**